packages/kit/types/page.d.ts

```diff
@@ -1,27 +1,42 @@
-export type LoadInput = {
+import { MaybePromise } from './helper';
+
+export type LoadInput<Context extends Record<string, any> = Record<string, any>> = {
 	page: Page;
 	fetch: (info: RequestInfo, init?: RequestInit) => Promise<Response>;
 	session: any;
-	context: Record<string, any>;
+	context: Context;
 };
 
-export type ErrorLoadInput = LoadInput & {
+export type ErrorLoadInput<
+	Context extends Record<string, any> = Record<string, any>
+> = LoadInput<Context> & {
 	status: number;
 	error: Error;
 };
 
-export type LoadOutput = {
+export type LoadOutput<
+	Props extends MaybePromise<Record<string, any>> = MaybePromise<Record<string, any>>,
+	Context extends Record<string, any> = Record<string, any>
+> = {
```

Comment:
Perhaps the `extends` could be omitted while still keeping the default values here, the users will only interact with `Load` anyway?

Reply:
I didn't realize these are not exported. I argue since they are indirectly public, they should be exposed, too.

Reply:
I can't think of any reason as to why anyone would need/want to use `LoadInput` and `LoadOutput` separately or individually, yet. But, I suppose it's better than having them hack it on their own using `Parameters` and `ReturnType`.